code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -117,6 +121,61 @@ func (sn *BlobberChallenge) GetHashBytes() []byte {
 }
 
 func (sn *BlobberChallenge) Decode(input []byte) error {
+	err := json.Unmarshal(input, sn)
+	if err != nil {
+		return err
+	}
+	sn.ChallengeIDMap = make(map[string]bool)
+	for _, challengeID := range sn.ChallengeIDs {
+		sn.ChallengeIDMap[challengeID] = true
+	}
+	return nil
+}
+
+func (sn *BlobberChallenge) addChallenge(challenge *StorageChallenge) bool {
+
+	if sn.ChallengeIDs == nil {
+		sn.ChallengeIDMap = make(map[string]bool)
+	}
+	if _, ok := sn.ChallengeIDMap[challenge.ID]; !ok {
+		if len(sn.ChallengeIDs) > 0 {
+			lastChallengeID := sn.ChallengeIDs[len(sn.ChallengeIDs)-1]
+			challenge.PrevID = lastChallengeID
+		} else if sn.LatestCompletedChallenge != nil {
+			challenge.PrevID = sn.LatestCompletedChallenge.ID
+		}
+		sn.ChallengeIDs = append(sn.ChallengeIDs, challenge.ID)
+		sn.ChallengeIDMap[challenge.ID] = true
+		return true
+	}
+	return false
+}
+
+type AllocationChallenge struct {
+	AllocationID             string                       `json:"allocation_id"`
+	Challenges               []*StorageChallenge          `json:"challenges"`
```

Comment:
This slice of `StorageChallenge` objects is too big. I think we need to come up with a way of avoiding needing it. In the meanwhile, the StoreaChallenger object only needs to know its Id and Created.